spec/rspec/core/example_group_spec.rb

```diff
@@ -140,6 +140,12 @@ def metadata_hash(*args)
         expect(child).to have_class_const("SomeParentGroup::Hash")
       end
 
+      it 'removes it\'s constants' do
+        groups = RSpec::ExampleGroups
+        groups.remove_all_constants
```

Comment:
That's to say, I'd remove this spec and add specs [here](https://github.com/rspec/rspec-core/blob/56e31e3a5cbeb296082e71e2fb46b6c98ef798a5/spec/rspec/core_spec.rb#L117-L127) and [here](https://github.com/rspec/rspec-core/blob/56e31e3a5cbeb296082e71e2fb46b6c98ef798a5/spec/rspec/core/world_spec.rb#L10-L16) instead.